packages/common/container-definitions/src/loader.ts

```diff
@@ -189,6 +189,7 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
 
     /**
      * Represents the resolved url to the Container
+     * TODO: what does it mean for this to be undefined?
```

Comment:
The resolved URL will only be undefined for a detached container -- that is, the service has no URL representing the container because it is unaware of the detached container.

Once the container becomes attached (or when loading from the service) we expect to acquire a URL associated with the container on the service.